roles/kubernetes-apps/gateway_api/defaults/main.yml

```diff
@@ -1,4 +1,4 @@
 ---
 gateway_api_enabled: false
-gateway_api_version: v1.1.0
+gateway_api_version: v1.2.0
```

Comment:
I wrote this code at the beginning, only change this is useless, in order to support offline installation, I finally did not use to this variable (PR #11376), I hope that the version can be differentiated by folder.

Reply:
Thanks for reminding. 
 1. I do not understand how the support of offline installation affects this variable.
 2. And how should I do to upgrade gateway_api. Should v1.1.0 be retained after updating?

Reply:
You can refer #11376.
My imagination is that you could put the original Gateway API CRDs in the `v1.1.0` folder, and you create a separate `v1.2.0` folder for the new version, and then you could use the `gateway_api_version`.

Reply:
IMO, difference gateway-api version by folder is not very good for scaling. We can't add a directory for every version because the list will be very long after later versions.

Reply:
Because Gateway API release cycle is not very frequent, reserve 3 patch version I think it's appropriate.
If you have a better idea, you can suggest in a comment.